mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -590,21 +588,17 @@ object NaiveBayesModel extends MLReadable[NaiveBayesModel] {
 
       instance.getModelType match {
         case Multinomial | Bernoulli | Complement =>
-          // Save model data: pi, theta
-          require(instance.sigma == null)
-          val data = Data(instance.pi, instance.theta)
-          sparkSession.createDataFrame(Seq(data)).repartition(1).write.parquet(dataPath)
-
+          require(instance.sigma.numRows == 0 && instance.sigma.numCols == 0)
```

Comment:
Would you get an NPE if loading a model saved with 2.4, where it's null? or am I missing that 2.4 would not have had this behavior (i.e. this is all new)?

Reply:
Yes, it is all new, the matrix `sigma` is just added in https://github.com/apache/spark/pull/26413 last week.